### DISCUSSION TITLE
Privacy modelling

### DISCUSSION THREAD

**Discussion by lvl99** [schemaorg/schemaorg#2867] [Ideas]
Hi all,

I've previously worked on some great projects which involved using Schema.org and it is a great resource for standardised data modelling.

One thing I've been thinking about recently in relation to GDPR and other e-Privacy directives (I'm based in Europe) is on how to describe an entity's attributes in relation to privacy.

For example, `givenName` and `familyName` are `Text` values, but their contents would be classified personal identifying information, so the integrating system may need to adjust how it handles/outputs the data based on the context of it being used.

I know the idea of building privacy by design into systems is kinda nascent (at least with regards to GDPR concepts like data portability and right to erasure/be forgotten) and I thought having some kind of standardised privacy annotation modelling of properties could be a worthwhile adventure, especially if it relates to standardised data entity models.

I'd propose that this kind of data modelling coupled with privacy annotations could allow not only the description of data's shape and format, but would also describe the handling of certain pieces of data in which integrated systems could then better treat certain pieces of information based on the context.

I was curious if anyone else has thought of this or would consider it something interesting to collaborate on?

**Comment by Guerreru** [comment]
Hi,

I am new to the group, after reading your notes, this is my understanding. This would be something that has to be done at the host/server level. As privacy rules apply at the information holders end. For example: if the givenName falls under the PII rules by GDPR, then you should omit it from the data disbursement and not be shared in any way publicly unless user has agree to do so. An example would be a jobs search website where the user might allow their Resume/Curriculum vitae to show in search results, and still the user has to opt in to display that certain data that falls under these rules. In other words, you cannot just disburse their givenName and add a note saying it falls under the GDPR rules, because then you or the company you work for will be breaking the GDPR rules for releasing the givenName.

Anyone please feel free to comment on this, if anything needs correction please specify. Thanks.

**Comment by thadguidry** [comment]
> I'd propose that this kind of data modelling coupled with privacy annotations could allow not only the description of data's shape and format, but would also describe the handling of certain pieces of data in which integrated systems could then better treat certain pieces of information based on the context.

That handling can already be done by systems themselves and so no need to add them to Schema.org or would we want to.
An integrated system would simply need to create their own annotations against Schema.org properties that THEY feel are privacy concerns in their respective jurisdictions and within their legal operating bounds in those jurisdictions.
Schema.org would never take a stance on which parts of structured data via Properties, Types, etc. might hold private pieces of data and might be a concern in someone's jurisdiction.
Those decisions are always best left up to each consuming application or organization or integrated system.

Having said that, if you are asking if Schema.org could or would provide a a simple way to annotate `<privateData>` somehow, DON'T.  The mere clear text transmission of unencrypted data will directly contradict all efforts to keep it private.  I.E. when you want to make something private, never use annotations, you will quickly attract evil-doers in seconds.  Instead, secure the data, encrypt it, etc. and use best practices.
